src/scripts/main.js

```diff
@@ -1,3 +1,74 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('.field');
+const rowButtons = document.querySelectorAll('[class*="row"]');
+const columnButtons = document.querySelectorAll('[class*="column"]');
+const MIN_PARTS = 2;
+const MAX_PARTS = 10;
+
+document.body.addEventListener('click', (e) => {
+  const button = e.target.closest('.button');
+
+  if (!button) {
+    return null;
+  }
+
+  const rows = [...table.querySelectorAll('tr')];
+  const column = rows[0].cells;
+  let rowsNumber = rows.length;
+  let columnNumber = column.length;
+  const rowProveRegex = /row/g;
+  const appendRegex = /append/g;
+  const buttonFunc = [...button.classList];
+  const containRows = buttonFunc.some((cls) => rowProveRegex.test(cls));
```

Comment:
```suggestion
  const containRows = button.classList.contains('append-row') || button.classList.contains('remove-row');
```

Reply:
Find how to put it in one line code =)

`const containRows = [...rowButtons].includes(button);`

Reply:
For example like that 
![image](https://github.com/mate-academy/js_growth_table_DOM/assets/78635631/9ef0df0b-e95a-4bf0-b01e-4499f5bc9851)